File: backend/components/metiers/administration_publique/tests/integration/test_threed.py
"""
test_threed.py - Tests unitaires Python pour Threed
"""
from ..fixtures import sample_3d_asset

def is_valid_model(model):
    return isinstance(model, dict) and 'id' in model and 'type' in model

def test_valid_3d_model():
    model = sample_3d_asset()
    assert is_valid_model(model)

def test_invalid_3d_model():
    assert not is_valid_model({})

File: backend/components/metiers/administration_publique/tests/integration/test_threed.js
// test_threed.js - Tests unitaires JS pour threed

const assert = require('assert');
const fixtures = require('../fixtures/fixtures');
const validator = require('../utils/validator');

describe('Threed Model Validation', () => {
  it('should validate a correct 3D model', () => {
    assert(validator.isValidModel(fixtures.sample3DModel));
  });

  it('should reject an invalid 3D model', () => {
    assert(!validator.isValidModel({}));
  });
});
